Solutions/Windows Security Events/Hunting Queries/Least_Common_Parent_Child_Process.yaml

```diff
@@ -14,22 +14,21 @@ requiredDataConnectors:
 tactics:
   - Execution
 query: |
-
   let starttime = todatetime('{{StartTimeISO}}');
   let endtime = todatetime('{{EndTimeISO}}');
   let lookback = starttime - 7d;
   let Allowlist = dynamic (['foo.exe', 'baz.exe']);
   let Sensitivity = 5;
   SecurityEvent
   | where TimeGenerated between(lookback..endtime)
-  | where EventID == 4688 and isnotnull(ParentProcessName)
+  | where EventID == 4688 and isnotempty(ParentProcessName)  
   | extend ProcArray = split(NewProcessName, '\\'), ParentProcArray = split(ParentProcessName, '\\')
   // ProcArrayLength is Folder Depth
-  | extend ProcArrayLength = arraylength(ProcArray), ParentProcArrayLength = arraylength(ParentProcArray)
+  | extend ProcArrayLength = array_length(ProcArray), ParentProcArrayLength = array_length(ParentProcArray)
   | extend LastIndex = ProcArrayLength - 1, ParentLastIndex = ParentProcArrayLength - 1
   | extend Proc = ProcArray[LastIndex], ParentProc = ParentProcArray[ParentLastIndex]
   | where Proc !in (Allowlist)
   | extend ParentChildPair = strcat(ParentProc , ' > ', Proc)
-  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), TimesSeen = count(), HostCount = dcount(Computer), Hosts = makeset(Computer), UserCount = dcount(SubjectUserName), Users = makeset(SubjectUserName) by ParentChildPair
+  | summarize StartTimeUtc = min(TimeGenerated), EndTimeUtc = max(TimeGenerated), TimesSeen = count(), HostCount = dcount(Computer), Hosts = make_set(Computer,maxSize=1000), UserCount = dcount(SubjectUserName), Users = make_set(SubjectUserName,maxSize=1000) by ParentChildPair
   | where TimesSeen < Sensitivity
   | extend timestamp = StartTimeUtc
```

Comment:
We should add entity mappings

Reply:
@petebryan : We are adding all hosts and user in an array, so that's why I think the mapping is not done